dali/python/nvidia/dali/plugin/numba/experimental.py

```diff
@@ -296,3 +298,136 @@ def run_cfunc(out_ptr, out_shapes_ptr, out_ndims_ptr, num_outs, in_ptr, in_shape
+        inputs = ops._preprocess_inputs(inputs, self._impl_name, self._device, None)
+        if pipeline is None:
+            Pipeline._raise_pipeline_required("NumbaFunction operator")
+        if (len(inputs) > self._schema.MaxNumInput() or
+                len(inputs) < self._schema.MinNumInput()):
+            raise ValueError(
+                ("Operator {} expects from {} to " +
+                 "{} inputs, but received {}.")
+                .format(type(self).__name__,
+                        self._schema.MinNumInput(),
+                        self._schema.MaxNumInput(),
+                        len(inputs)))
+        for inp in inputs:
+            if not isinstance(inp, _DataNode):
+                raise TypeError(
+                      ("Expected inputs of type `DataNode`. Received input of type '{}'. " +
+                       "Python Operators do not support Multiple Input Sets.")
+                      .format(type(inp).__name__))
+        op_instance = ops._OperatorInstance(inputs, self, **kwargs)
+        op_instance.spec.AddArg("run_fn", self.run_fn)
+        if self.setup_fn != None:
+            op_instance.spec.AddArg("setup_fn", self.setup_fn)
+        op_instance.spec.AddArg("out_types", self.out_types)
+        op_instance.spec.AddArg("in_types", self.in_types)
+        op_instance.spec.AddArg("outs_ndim", self.outs_ndim)
+        op_instance.spec.AddArg("ins_ndim", self.ins_ndim)
+        op_instance.spec.AddArg("blocks", self.blocks)
+        op_instance.spec.AddArg("threads_per_block", self.threads_per_block)
+        op_instance.spec.AddArg("device", self.device)
+        op_instance.spec.AddArg("batch_processing", self.batch_processing)
+
+        if self.num_outputs == 0:
+            t_name = self._impl_name + "_id_" + str(op_instance.id) + "_sink"
+            t = _DataNode(t_name, self._device, op_instance)
+            pipeline.add_sink(t)
+            return
+        outputs = []
+
+
+        for i in range(self.num_outputs):
+            t_name = op_instance._name
+            if self.num_outputs > 1:
+                t_name += "[{}]".format(i)
+            t = _DataNode(t_name, self._device, op_instance)
+            op_instance.spec.AddOutput(t.name, t.device)
+            op_instance.append_output(t)
+            pipeline.add_sink(t)
+            outputs.append(t)
+        return outputs[0] if len(outputs) == 1 else outputs
+
+    def __init__(self, run_fn, out_types, in_types, outs_ndim, ins_ndim, blocks, threads_per_block, setup_fn=None, device='cpu', batch_processing=False, **kwargs):
+        assert len(in_types) == len(ins_ndim), "Number of input types and input dimensions should match."
+        assert len(out_types) == len(outs_ndim), "Number of output types and output dimensions should match."
+        assert len(blocks) == 3, "`blocks` should be an array of 3 numbers."
+        assert len(threads_per_block) == 3, "`threads_per_block` should be an array of 3 numbers."
```

Comment:
Maybe you can assert if all elements are `>=0` as well?

Reply:
Done